packages/ragbits-core/src/ragbits/core/audit/cli.py

```diff
@@ -0,0 +1,132 @@
+import time
+from typing import Optional
+
+from rich import print as rich_print
+from rich.tree import Tree
+
+from ragbits.core.audit import TraceHandler
+
+
+class CLISpan:
+    """
+    CLI Span represents a single operation within a trace.
+    """
+
+    def __init__(self, name: str, inputs: dict, parent: Optional["CLISpan"] = None):
+        """
+        Constructs a new CLI Span.
+        Sets the start time of the span - the wall time at which the operation started.
+        Sets the span status to 'started'.
+
+        Args:
+            name: The name of the span.
+            inputs: The inputs of the span.
+            parent: the parent of initiated span.
+        """
+        self.name = name
+        self.parent = parent
+        self.start_time: float = time.perf_counter()
+        self.end_time: float | None = None
+        self.children: list[CLISpan] = []
+        self.status: str = "started"
+        self.inputs: dict = inputs or {}
+        self.outputs: dict = {}
+
+    def end(self) -> None:
+        """Sets the current time as the span's end time.
+        The span's end time is the wall time at which the operation finished.
+        Only the first call to `end` should modify the span,
+        further calls are ignored.
+        """
+        if self.end_time is None:
+            self.end_time = time.perf_counter()
+
+    def to_tree(self, tree: Tree | None = None, color: str = "bold blue") -> Tree | None:
+        """
+        Convert theCLISpan object and its children into a Rich Tree structure for console rendering.
+
+        Args:
+            tree (Tree, optional): An existing Rich Tree object to which the span will be added.
+                               If None, a new tree is created for the root span.
+            color (str, optional): The color of the text rendered to console.
+
+        Returns:
+            Tree: A Rich Tree object representing the span hierarchy, including its events and children.
+        """
+        secondary_color = "grey50"
+        error_color = "bold red"
+        child_color = "bold green"
+        duration = self.end_time - self.start_time if self.end_time else 0.0
+
+        if tree is None:
+            tree = Tree(
+                f"[{color}]{self.name}[/{color}] Duration: {duration:.3f}s\n"
+                f"[{secondary_color}]Inputs: {self.inputs}\nOutputs: {self.outputs})[/{secondary_color}]"
+            )
+
+        else:
+            child_tree = tree.add(
+                f"[{color}]{self.name}[/{color}] Duration: {duration:.3f}s\n"
+                f"[{secondary_color}]Inputs: {self.inputs}\nOutputs: {self.outputs})[/{secondary_color}]"
```

Comment:
maybe we can make this more readable, lets try to print each input/output key separately and use different colors for keys and values

Reply:
we can also add some parameters to the `CLITraceHandler` to control whether we want the input/output data to be printed + optional truncation

Reply:
I've added printing input/output keys separately and truncation of the long values. I haven't add the parameter for optional input/output printing.